app/components/home/GuestHome/Dialog/signUp/SignUpForm.jsx

```diff
@@ -0,0 +1,220 @@
+import React from 'react';
+import FlatButton from 'material-ui/FlatButton';
+import TextField from 'material-ui/TextField';
+import Snackbar from 'material-ui/Snackbar';
+import request from 'superagent';
+
+const divStyle = {
+  marginTop: '20px',
+};
+
+const fieldStyle = {
+  display: 'block',
+  margin: 'auto',
+  width: '75%',
+};
+
+const buttonStyle = {
+  display: 'block',
+  margin: 'auto',
+  width: '75%',
+  color: '#FFFFFF',
+  backgroundColor: '#607D8B',
+  marginTop: '20px',
+};
+
+class SignUpForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      snackBarOpen: false,
+      username: '',
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+      disableButton: true,
+      errorText: ''
+    };
+
+    this.formReady = this.formReady.bind(this);
+    this.signUpUser = this.signUpUser.bind(this);
+  }
+
+  formReady() {
+    const ready =
+      this.isAlphanumeric(this.state.username).length < 1 &&
+      this.isAlphanumeric(this.state.firstName).length < 1 &&
+      this.isAlphanumeric(this.state.lastName).length < 1 &&
+      this.isEmail(this.state.email).length < 1 &&
+      this.state.password.length > 0;
+    this.setState({ disableButton: !ready });
+  }
+
+  isAlphanumeric(value) {
+    return /^[a-zA-Z0-9_]/.test(value) ?
+      '' :
+      'This value must be alphanumeric';
+  }
+
+  isEmail(value) {
+    return /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value) ?
+      '' :
+      'A valid email is required';
+  }
+
+  validate(field) {
+    switch (field) {
+      case ('username'):
+        return this.state.username ?
+          this.isAlphanumeric(this.state.username) :
+          'This field is required';
+      case ('firstName'):
+        return this.state.firstName ?
+          this.isAlphanumeric(this.state.firstName) :
+          'This field is required';
+      case ('lastName'):
+        return this.state.lastName ?
+          this.isAlphanumeric(this.state.lastName) :
+          'This field is required';
+      case ('email'):
+        return this.state.email ?
+          this.isEmail(this.state.email) :
+          'This field is required';
+      case ('password'):
+        return this.state.password ?
+          '' :
+          'This field is required';
+      default:
+        return '';
+    }
+  }
+  signUpUser() {
+    request
```

Comment:
**Code  Quality - Critical**
If you're using Redux, you should consider moving this (and any other API calls) into either actions or middleware. Makes your code easier to reason about.